app/controllers/peer_reviews_controller.rb

```diff
@@ -4,9 +4,12 @@ class PeerReviewsController < ApplicationController
 
   before_action :set_peer_review, only: [:show, :edit, :update, :destroy]
 
-  before_action :authorize_only_for_admin, except: [:list_reviews, :show_reviews, :show_result]
+  before_action :authorize_only_for_admin, except: [:list_reviews, :index, :populate, :assign_groups,
+                                                    :show_reviews, :show_result]
   before_action :authorize_for_user, only: [:list_reviews, :show_reviews, :show_result]
-
+  before_action only: [:index, :populate, :assign_groups] do
```

Comment:
It's actually okay to authorize all methods here except for the original three exceptions. So you can do the same `before_action except:`, but now call `authorize!` instead of `authorize_only_for_admin`.